site/src/pages/docs/advanced-metrics.mdx

```diff
@@ -0,0 +1,73 @@
+# Collecting metrics
+
+Armeria has built-in support for collecting metrics both on the server and client side. This page describes how to enable this.
+
+The metric data is collected using the [Micrometer](https://micrometer.io/) library. Please consult its documentation for information on how to expose the collected metrics to various monitoring systems (JMX, Prometheus, etc)
+
+## Collecting server-side metrics (gRPC)
+
+```java
+import com.linecorp.armeria.common.grpc.GrpcMeterIdPrefixFunction;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.grpc.GrpcService;
+import com.linecorp.armeria.server.metric.MetricCollectingService;
+
+ServerBuilder serverBuilder = Server.builder()
+        .http(new InetSocketAddress(address, port))
+        .service(
```

Comment:
How about this style for readability?
```suggestion
ServerBuilder serverBuilder = 
        Server.builder()
              .http(new InetSocketAddress(address, port))
              .service(
```

Reply:
Can do. The main problem I have with the above is that I cannot auto-format the code (using IntelliJ) to it. Is there an `.editorconfig` or similar that can help me with that?

Reply:
Armeria does not have `.editorconfig` because
- For Java, use [checkstyle](https://checkstyle.sourceforge.io/) and [a custom formatter](https://armeria.dev/community/developer-guide#setting-up-your-ide)
- For Typescript, use [eslintrc](https://github.com/line/armeria/blob/7f89c81dab44931764bf943910190cd1cf72f643/settings/eslint/eslintrc.js)

IntelliJ supports [MDX plugin](https://plugins.jetbrains.com/plugin/14944-mdx) since 2020.2 version. However, the plugin cannot not format code block automatically.

I think it needs a human touch at the monment. If you don't mind, I can help your code format instead of a machine before merging this PR. 😉

Reply:
Oh we have a plan to add it.
https://github.com/line/armeria/pull/3164#discussion_r519617291

Reply:
@ikhoon Yes, feel free to add a commit that fixes the code formatting, wouldn't mind.

Reply:
Sure, I will format the code of this documentation when other comments are addressed.